src/formatDate.js

```diff
@@ -9,6 +9,41 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const splitedDate = date.split(oldSeparator);
+
+  const newDate = [];
+
+  for (let i = 0; i < splitedDate.length; i++) {
+    const oldPart = fromFormat[i];
+
+    for (let j = 0; j < splitedDate.length; j++) {
+      const newPart = toFormat[j];
+
+      if (oldPart[0] === newPart[0]) {
+        let newDateElement = splitedDate[i];
+
+        newDateElement = splitedDate[i];
+
+        if (oldPart.length > newPart.length) {
+          newDateElement = newDateElement.split('').slice(-2).join('');
+        }
+
+        if (oldPart.length < newPart.length) {
+          newDateElement =
+            +newDateElement < 30
+              ? `20${newDateElement}`
+              : `19${newDateElement}`;
+        }
+
+        newDate[j] = newDateElement;
+      }
+    }
```

Comment:
The nested loop is unnecessary and will lead to incorrect results because it tries to match each 'fromFormat' part with every 'toFormat' part, which isn't the correct logic. You should directly map each part of the date to its new format based on the index. For example, you can create a map of indexes from the old format to the new format and then rebuild the date string accordingly.